### PR TITLE
Refactor and add workflow for generate projects page

### LINKED ISSUE #4: Set up GitHub Actions to run the generate project pages script every day
## Problem Description
We currently use AWS Lambda to run the generate project page script, but it's set up manually and is probably unnecessary. It's also hard to add new dependencies to the environment.

We do want the script run every day so that the projects page is always up to date.

## Solution Description
Set up a GitHub Actions workflow in this repo to run the script every day.

## Additional context
- [GitHub Actions documentation](https://help.github.com/en/actions)
- You will need to remove the dependency on `boto3` since that's only used on AWS Lambda.
- You should probably replace the `lambda-git` package that's used with something else.
- Please include instructions on how to set up the appropriate GitHub token. Your GitHub token will not have permissions to commit to the existing https://github.com/creativecommons/creativecommons.github.io-source repo, so you'll need to test using a fork of the repo on your own account.

### PR DESCRIPTION
**Pull request by sp35**
Fixes #4 

## Description
Used GitPython https://pypi.org/project/GitPython/ to replace lambda-git
Added workflow for the same

## Other information
Steps to generate and setup token:
Generate a token with at least the following scopes:
a. ```repo```  (as of now)
Add this token to the secret of the repo with the name 'ADMIN_GITHUB_TOKEN'

To test the workflow:
```
python -m pip install --upgrade pip
python -m pip install pipenv
cd generate_projects_page
pipenv install
export GITHUB_TOKEN=<ADMIN_GITHUB_TOKEN>
python3 generate_projects_page.py
```

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

### CODE REVIEW

File: .github/workflows/generate_projects_page.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        python -m pip install --upgrade pip
+        python -m pip install pipenv
+        cd generate_projects_page
+        pipenv install
+    - name: Export token to env and run the script
+      run: |
+        cd generate_projects_page
+        pipenv run python generate_projects_page.py
+      env:
+        GEN_PAGE_GITHUB_TOKEN: ${{ secrets.GEN_PAGE_GITHUB_TOKEN }}
```
Comment (kgodey) [inline .github/workflows/generate_projects_page.yml]: could we use `ADMIN_GITHUB_TOKEN` here? we already have it set up on the repo for the other script.

Comment (sp35) [reply .github/workflows/generate_projects_page.yml]: Amended the change

**Comment by sp35** [comment]
@TimidRobot Sir, would you like to review?
I can see an 'On vacation until Mar 10' status by @kgodey Ma'am.

File: generate_projects_page/generate_projects_page.py [hunk 2]
```diff
@@ -1,57 +1,53 @@
...
 
-LAMBDA_WORKING_DIRECTORY = '/tmp'
-GIT_WORKING_DIRECTORY = f'{LAMBDA_WORKING_DIRECTORY}/{GITHUB_REPO_NAME}'
-JSON_FILE_DIRECTORY = f'{GIT_WORKING_DIRECTORY}/databags'
+
+GITHUB_TOKEN = os.environ["ADMIN_GITHUB_TOKEN"]
+GITHUB_REPO_URL_WITH_CREDENTIALS = f"https://{GITHUB_USERNAME}:{GITHUB_TOKEN}@github.com/{GITHUB_ORGANIZATION}/{GITHUB_REPO_NAME}.git"
+CC_METADATA_FILE_NAME = ".cc-metadata.yml"
+
+LAMBDA_WORKING_DIRECTORY = "/tmp"
```
Comment (kgodey) [inline generate_projects_page/generate_projects_page.py]: We should rename this since we're not using Lambda.

**Comment by sp35** [comment]
Yes, the script compiles locally and I have set-up the workflow under my fork.
Regarding testing by committing to my fork of https://github.com/creativecommons/creativecommons.github.io-source, that's pending as I'm not quite sure of how to proceed.

**Comment by kgodey** [comment]
I'll merge and see if it works.

**Pull request merged**